Enable HIP build for //sigrid/predictor:pytorch_disagg_gpu_task

houseroad opened the pull request:
Summary: Tweak some header include, as well as explicitly ignore hipEventDestroy return value.

Test Plan: CI

Reviewed By: jiaqizhai

Differential Revision: D52722234

xw285cornell commented on aten/src/ATen/cuda/CUDAEvent.h:
```diff
@@ -50,7 +50,8 @@ struct TORCH_CUDA_CPP_API CUDAEvent {
         if (C10_UNLIKELY(interp)) {
           (*interp)->trace_gpu_event_deletion(reinterpret_cast<uintptr_t>(event_));
         }
-        cudaEventDestroy(event_);
```
I wonder if we can do this? 
`
AT_CUDA_CHECK(cudaEventDestroy(event_));
`

houseroad committed:
Enable HIP build for //sigrid/predictor:pytorch_disagg_gpu_task (#117616)

Summary:

Tweak some header include, as well as explicitly ignore hipEventDestroy return value.

Test Plan:
OSS:

CI

Reviewed By: jiaqizhai, xw285cornell

Differential Revision: D52722234
